File: C++/Hard/WrongAnswer/number-of-distinct-roll-sequences_7.cpp
class Solution {
public:
    int distinctSequences(int n) {
        
        if (n == 1) return 6;
        if (n == 2) return 22;
        vector<vector<int>> dp(n + 1, vector<int>(6, 0));
        for (int i = 0; i < 6; i++) dp[1][i] = 1;
        for (int i = 2; i <= n; i++) {
            for (int j = 0; j < 6; j++) {
                for (int k = 0; k < 6; k++) {
                    if (j != k && abs(j - k) > 1) {
                        dp[i][j] += dp[i - 1][k];
                        dp[i][j] %= 1000000007;
                    }
                }
            }
        }
        int res = 0;
        for (int i = 0; i < 6; i++) {
            res += dp[n][i];
            res %= 1000000007;
        }
        return res;
    }
};

File: Java/Hard/CompileError/number-of-distinct-roll-sequences_7.java
class Solution {
    public int distinctSequences(int n) {
        
        int mod = 1000000007;
        long[][] dp = new long[n+1][7];
        for(int i = 1; i <= 6; i++) dp[1][i] = 1;
        for(int i = 2; i <= n; i++) {
            for(int j = 1; j <= 6; j++) {
                for(int k = 1; k <= 6; k++) {
                    if(j == k) continue;
                    if(gcd(j, k) > 1) continue;
                    dp[i][j] += dp[i-1][k];
                    dp[i][j] %= mod;
                }
            }
        }
        long ans = 0;
        for(int i = 1; i <= 6; i++) {
            ans += dp[n][i];
            ans %= mod;
        }
        return (int)ans;
    }
}